LocalPackages/Freemium/Tests/FreemiumTests/FreemiumPIRUserStateManagerTests.swift

```diff
@@ -84,4 +88,118 @@ final class FreemiumPIRUserStateManagerTests: XCTestCase {
+        XCTAssertFalse(testUserDefaults.bool(forKey: Keys.didPostResultsNotification))
+
+        // When
+        sut.didPostResultsNotification = true
+
+        // Then
+        XCTAssertTrue(testUserDefaults.bool(forKey: Keys.didPostResultsNotification))
+    }
+
+    func testGetsDidPostResultsNotification() throws {
+        // Given
+        let sut = DefaultFreemiumPIRUserStateManager(userDefaults: testUserDefaults)
+        XCTAssertFalse(sut.didPostResultsNotification)
+        testUserDefaults.setValue(true, forKey: Keys.didPostResultsNotification)
+        XCTAssertTrue(testUserDefaults.bool(forKey: Keys.didPostResultsNotification))
+
+        // When
+        let result = sut.didPostResultsNotification
+
+        // Then
+        XCTAssertTrue(result)
+    }
+
+    func testSetsDidDismissHomePagePromotion() throws {
+        // Given
+        let sut = DefaultFreemiumPIRUserStateManager(userDefaults: testUserDefaults)
+        XCTAssertFalse(testUserDefaults.bool(forKey: Keys.didDismissHomePagePromotion))
+
+        // When
+        sut.didDismissHomePagePromotion = true
+
+        // Then
+        XCTAssertTrue(testUserDefaults.bool(forKey: Keys.didDismissHomePagePromotion))
+    }
+
+    func testGetsDidDismissHomePagePromotion() throws {
+        // Given
+        let sut = DefaultFreemiumPIRUserStateManager(userDefaults: testUserDefaults)
+        XCTAssertFalse(sut.didDismissHomePagePromotion)
+        testUserDefaults.setValue(true, forKey: Keys.didDismissHomePagePromotion)
+        XCTAssertTrue(testUserDefaults.bool(forKey: Keys.didDismissHomePagePromotion))
+
+        // When
+        let result = sut.didDismissHomePagePromotion
+
+        // Then
+        XCTAssertTrue(result)
+    }
+
+    func testSetsFirstScanResults() throws {
+        // Given
+        let sut = DefaultFreemiumPIRUserStateManager(userDefaults: testUserDefaults)
+        XCTAssertNil(testUserDefaults.data(forKey: Keys.firstScanResults))
+
+        // When
+        let scanResults = FreemiumDBPMatchResults(matchesCount: 3, brokerCount: 2)
+        sut.firstScanResults = scanResults
+
+        // Then
+        let storedData = testUserDefaults.data(forKey: Keys.firstScanResults)
+        XCTAssertNotNil(storedData)
+
+        // Decode and verify the result
+        let decodedResults = try? JSONDecoder().decode(FreemiumDBPMatchResults.self, from: storedData!)
+        XCTAssertEqual(decodedResults?.matchesCount, scanResults.matchesCount)
+        XCTAssertEqual(decodedResults?.brokerCount, scanResults.brokerCount)
+    }
+
+    func testGetsFirstScanResults() throws {
+        // Given
+        let sut = DefaultFreemiumPIRUserStateManager(userDefaults: testUserDefaults)
+        XCTAssertNil(sut.firstScanResults)
+
+        // When
+        let scanResults = FreemiumDBPMatchResults(matchesCount: 3, brokerCount: 2)
+        let encodedResults = try! JSONEncoder().encode(scanResults)
```

Comment:
Swiftlint can become annoying but there's no reason to force unwrap it in unit tests because the test itself is a throwing function. Worst case the test will be marked as failing, and with force unwrapping the test will crash here and stop executing the rest 👍